shm/map.cpp

```diff
@@ -0,0 +1,10 @@
+#include "map.hpp"
+
+#include <algorithm>
+
+Island* Map::getIsland(const Coordinates& coordinate) {
+    auto result = std::find_if(Islands_.begin(), Islands_.end(), [coordinate](const Island& islnd) {
+        return islnd.getCoordinates() == coordinate;
+    });
+    return &(*result);
+}
```

Comment:
It looks strange but don't know if can be done better.
For me is fine.
And new line at the end of file.